runtime/jvmti/jvmtiHook.c

```diff
@@ -1112,7 +1112,12 @@ findFieldIndexFromOffset(J9VMThread *currentThread, J9Class *clazz, UDATA offset
 		J9ROMClass * const romClass = clazz->romClass;
 		J9Class * const superclazz = GET_SUPERCLASS(clazz);
 		J9ROMFieldOffsetWalkState state;
-		J9ROMFieldOffsetWalkResult *result = vmFuncs->fieldOffsetsStartDo(vm, romClass, superclazz, &state, walkFlags);
+		J9ROMFieldOffsetWalkResult *result = NULL;
+
+		state.classLoader = clazz->classLoader;
```

Comment:
As usual, I'm flip-flopping here. If this point is guaranteed to not need the loader, let's set it to NULL and assert in the offsets code is you need to load something and the loader has not been provided.

The callers of this code can not handle java code running, so let's make the NULL be the indicator for that.

Reply:
I'm wrong - we always need the loader, but we'll be passing it as a parameter to the function (see below).